# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: e72ef41a4a78
# Full Hash: e72ef41a4a781d9f754588d5076a4e4bf58008b6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-10 15:51:07
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Added documentation on Cursor data members. r=ttung
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D41403
# ==============================================================================

diff -r 606b15c63139 -r e72ef41a4a78 dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Tue Sep 10 14:31:47 2019 +0100
+++ b/dom/indexedDB/ActorsParent.cpp	Tue Sep 10 16:46:44 2019 +0300
@@ -7526,10 +7526,28 @@
   nsCString mContinuePrimaryKeyQuery;
   nsCString mLocale;
 
-  Key mKey;
-  Key mObjectKey;
-  Key mRangeKey;
-  Key mSortKey;
+  // TODO: Apply the renamings suggested below, and also change related
+  // identifiers (e.g. local variables) and literals.
+
+  Key mKey;  ///< The current key, i.e. the key representing the cursor's
+             ///< position (https://w3c.github.io/IndexedDB/#cursor-position).
+             ///<
+             ///< TODO: Rename this to mPosition
+  Key mObjectKey;  ///< The key representing the cursor's object store position
+  ///< (https://w3c.github.io/IndexedDB/#cursor-object-store-position).
+  ///<
+  ///< TODO: Rename this to mObjectStorePosition
+  Key mRangeKey;  ///< If the cursor is based on a key range, the bound in the
+                  ///< direction of iteration (e.g. the upper bound in case of
+                  ///< mDirection == NEXT). If the cursor is based on a key, it
+                  ///< is unset. If mLocale is set, this was converted to
+                  ///< mLocale.
+                  ///<
+                  ///< TODO: Rename this to mLocaleAwareRangeBound
+  Key mSortKey;   ///< If mLocale is set, this is mKey converted to mLocale.
+                  ///< Otherwise, it is unset.
+                  ///<
+                  ///< TODO: Rename this to mLocaleAwarePosition
 
   CursorOpBase* mCurrentlyRunningOp;
 
